lib/common/socket.c

```diff
@@ -977,9 +977,9 @@ void h2o_socket_ssl_handshake(h2o_socket_t *sock, SSL_CTX *ssl_ctx, const char *
     } else {
         h2o_cache_t *session_cache = h2o_socket_ssl_get_session_cache(ssl_ctx);
         if (session_cache != NULL) {
-            struct sockaddr sa;
+            struct sockaddr_storage sa;
             int32_t port;
-            if (h2o_socket_getpeername(sock, &sa) != 0 && (port = h2o_socket_getport(&sa)) != -1) {
+            if (h2o_socket_getpeername(sock, (struct sockaddr *)&sa) != 0 && (port = h2o_socket_getport((struct sockaddr *)&sa)) != -1) {
```

Comment:
Off topic, but we might consider changing the signature of `h2o_socket_getpeername` to take three arguments instead of two, following the API of `getsockname`.